CONTRIBUTING.md

```diff
@@ -39,6 +41,51 @@ To develop the `docs/` site:
 
 2. Run `pnpm run docs` in Vite's root folder.
 
+### Docs Translation Contribution
+
+To add a new language to the Vite docs, see [`vite-docs-template`](https://github.com/tony19/vite-docs-template/blob/main/.github/CONTRIBUTING.md).
+
+## Notes on Dependencies
+
+Vite aims to be lightweight, and this includes being aware of the number of npm dependencies and their size.
+
+We use Rollup to pre-bundle most dependencies before publishing! Therefore, most dependencies, even those used in runtime source code, should be added under `devDependencies` by default. This also creates the following constraints that we need to be aware of in the codebase.
+
+### Usage of `require()`
+
+In some cases, we intentionally lazy-require some dependencies to improve start-up performance. However, note that we cannot use simple `require('somedep')` calls since these are ignored in ESM files, so the dependency won't be included in the bundle, and the actual dependency won't even be there when published since they are in `devDependencies`.
+
+Instead, use `(await import('somedep')).default`.
+
+### Think Before Adding a Dependency
+
+Most deps should be added to `devDependencies` even if they are needed at runtime. Some exceptions are:
+
+- Type packages. Example: `@types/*`.
+- Deps that cannot be properly bundled due to binary files. Example: `esbuild`.
+- Deps that ship their own types that are used in Vite's own public types. Example: `rollup`.
+
+Avoid deps with large transitive dependencies that result in bloated size compared to the functionality it provides. For example, `http-proxy` itself is around 380kB in size, but `http-proxy-middleware` pulls in a ton of dependencies that make it 3MB(!) when a minimal custom middleware on top of `http-proxy` only requires a couple of lines of code.
+
+### Ensure Type Support
+
+Vite aims to be fully usable as a dependency in a TypeScript project (e.g. it should provide proper typings for VitePress), and also in `vite.config.ts`. This means technically a dependency whose types are exposed needs to be part of `dependencies` instead of `devDependencies`. However, this also means we won't be able to bundle it.
+
+To get around this, we inline some of these dependencies' types in `packages/vite/src/types`. This way, we can still expose the typing but bundle the dependency's source code.
+
+Use `pnpm run build-types-check` to check that the bundled types do not rely on types in `devDependencies`.
+
+For types shared between client and node, they should be added into `packages/vite/types`. These types are not bundled and are published as is (though they are still considered internal).
```

Comment:
Removed "Dependency types within this directory (e.g. `packages/vite/types/chokidar.d.ts`) are deprecated and should be added to `packages/vite/src/types` instead.". These types no longer exists.